bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/ScalingSWTFontRegistry.java

```diff
@@ -44,7 +44,9 @@ private Font getScaledFont(int zoom) {
 
 		private Font createAndCacheFont(int zoom) {
 			Font newFont = createFont(zoom);
-			customFontHandlesKeyMap.put(Font.win32_getHandle(newFont), this);
+			FontData clonedFontData = new FontData(newFont.getFontData()[0].toString());
```

Comment:
You may adapt this to the new copy constructor since https://github.com/eclipse-platform/eclipse.platform.swt/pull/2128 is merged now.